sqlglot/dialects/tsql.py

```diff
@@ -328,6 +328,21 @@ def _json_extract_sql(
     return self.func("ISNULL", json_query, json_value)
 
 
+def _replace_limit_with_offset_fetch(expression: exp.Expression) -> exp.Expression:
```

Comment:
doesn't this happen in the generator? why do we need this?

Reply:
For context, `TOP` can't be used with `OFFSET`, so we need to produce the `FETCH` clause if we need to offset the rowset.